src/assembler.jl

```diff
@@ -209,3 +209,66 @@ function InsertionSort!(A, order, ii=1, jj=length(A))
     end  # i
     return
 end
+
+
+struct LinearAssembler{Tv,Ti}
+    K::SparseMatrixCSC{Tv,Ti}
+    dict::Dict{Tuple{Ti,Ti},Ti}
+end
+
+function LinearAssembler(dh::DofHandler)
+    K, d = _create_linear_sparsity_pattern(dh,false)
+    return LinearAssembler(K,d)
+end
+
+function _create_linear_sparsity_pattern(dh::DofHandler, sym::Bool)
+    ncells = getncells(dh.grid)
+    n = ndofs_per_cell(dh)
+    N = sym ? div(n*(n+1), 2) * ncells : n^2 * ncells
+    N += ndofs(dh) # always add the diagonal elements
+    I = Int[]; sizehint!(I, N)
+    J = Int[]; sizehint!(J, N)
+    V = Tuple{Int,Int}[]; sizehint!(V, N)
+    global_dofs = zeros(Int, n)
+    for element_id in 1:ncells
+        celldofs!(global_dofs, dh, element_id)
+        @inbounds for j in 1:n, i in 1:n
+            dofi = global_dofs[i]
+            dofj = global_dofs[j]
+            sym && (dofi > dofj && continue)
+            push!(I, dofi)
+            push!(J, dofj)
+            push!(V, (dofi, dofj))
+        end
+    end
+    for d in 1:ndofs(dh)
+        push!(I, d)
+        push!(J, d)
+        push!(V, (d,d))
+    end
+    K = sparse(I, J, V, maximum(I), maximum(J), (x,y) -> x)
+
+    # loop again, and construct the map from (dofi, dofj)
+    # to a linear index into K.nzval
+    dict = Dict{Tuple{Int,Int},Int}()
+    for element_id in 1:ncells
```

Comment:
Why a dict and not a matrix?

Reply:
Hmm... Interesting. A SparseMatrix that gives the index to another one then? :)

Reply:
But then this has no advantage over simply `K[i,j] += ...`?

Reply:
You don't have to binary search?

Reply:
You don't think a hash lookup in a dict is faster than finding the correct index in a SparseMatrixCSC?